src/psyclone/psyir/backend/fortran.py

```diff
@@ -192,6 +202,67 @@ def gen_declaration(self, symbol):
         result += "\n"
         return result
 
+    def container_node(self, node):
+        '''This method is called when a Container instance is found in
+        the PSyIR tree.
+
+        A container node is mapped to a module in the Fortran back end.
+
+        :param node: a Container PSyIR node.
+        :type node: :py:class:`psyclone.psyGen.Container`
+
+        :returns: the Fortran code as a string.
+        :rtype: str
+
+        :raises VisitorError: if the name attribute of the supplied \
+        node is empty or None.
+
+        '''
+        if not node.name:
+            raise VisitorError("Expected node name to have a value.")
+
+        # All children must be kernelschedule as modules within
+        # modules are not supported.
+        from psyclone.psyGen import KernelSchedule
+        if not all([isinstance(child, KernelSchedule)
+                    for child in node.children]):
+            raise VisitorError(
+                "The Fortran back-end requires all children of a Container "
+                "to be KernelSchedules.")
+
+        result = (
+            "{0}module {1}\n"
```

Comment:
You could move this line up onto the previous one.

Reply:
Oh yes. I think the statement was a lot longer previously :-).